lib/src/line_to_polygon.dart

```diff
@@ -0,0 +1,181 @@
+  } else {
+    throw exc;
+  }
+  if (!mutate) {
+    lines = lines.clone();
+  }
+
+  if (lines is FeatureCollection) {
+    List<List<List<Position>>> coords = [];
+    featureEach(
+      lines,
+      ((line, featureIndex) => coords.add(getCoords(lineStringToPolygon(
+              line, autoComplete, orderCoords, properties: {}))
+          as List<List<Position>>)),
+    );
+    return Feature(
+        geometry: MultiPolygon(coordinates: coords), properties: properties);
+  } else {
+    return lineStringToPolygon(lines, autoComplete, orderCoords,
+        properties: properties);
+  }
+}
+
+/// Converts LineString to Polygon
+/// Takes a optional bool autoComplete=true that auto completes linestrings
+/// Takes an optional orderCoords=true that sorts linestrings to place outer
+/// ring at the first position of the coordinates.
+Feature<Polygon> lineStringToPolygon(
+    GeoJSONObject line, bool autoComplete, bool orderCoords,
+    {Map<String, dynamic>? properties}) {
+  properties = properties ?? (line is Feature ? line.properties ?? {} : {});
+  var geom = line is LineString ? line : (line as Feature).geometry;
+  List<dynamic> coords = (geom is LineString || geom is MultiLineString)
+      ? (geom is LineString)
+          ? geom.coordinates
+          : (geom as MultiLineString).coordinates
+      : ((geom as Feature).geometry as GeometryType).coordinates;
+
+  if (coords.isEmpty) throw Exception("line must contain coordinates");
+
+  if (geom is LineString) {
+    if (autoComplete) {
+      coords = _autoCompleteCoords(coords as List<Position>);
+    }
+    return Feature(
+        geometry: Polygon(coordinates: [coords as List<Position>]),
+        properties: properties);
+  } else if (geom is MultiLineString) {
+    List<List<Position>> multiCoords = [];
+    num largestArea = 0;
+
+    (coords as List<List<Position>>).forEach((coord) {
+      if (autoComplete) {
+        coord = _autoCompleteCoords(coord);
+      }
+
+      // Largest LineString to be placed in the first position of the coordinates array
+      if (orderCoords) {
+        var area = _calculateArea(bbox(LineString(coordinates: coord)));
+        if (area > largestArea) {
+          multiCoords.insert(0, coord);
+          largestArea = area;
+        } else {
+          multiCoords.add(coord);
+        }
+      } else {
+        multiCoords.add(coord);
+      }
+    });
+    return Feature(
+        geometry: Polygon(coordinates: multiCoords), properties: properties);
+  } else {
+    throw Exception(
+        "geometry type  ${(geom as GeoJSONObject).type}  is not supported");
+  }
+}
+
+/// Auto Completes Coords - matches first & last coordinates
+List<Position> _autoCompleteCoords(List<Position> coords) {
+  var first = coords[0];
+  var x1 = first[0];
+  var y1 = first[1];
+  var last = coords[coords.length - 1];
+  var x2 = last[0];
+  var y2 = last[1];
+  if (x1 != x2 || y1 != y2) {
+    coords.add(first);
+  }
+  return coords;
```

Comment:
It is very important to **always** clone a class, otherwise it is a **reference**, not a newly instantiated object...

you can simplify to a great extent here:
```dart
    var newCoords = coords.map((c) => c.clone()).toList();
    if(newCoords.first != newCoords.last){
        newCoords.add(first.clone());
    }
    return newCoords;
```